internal/db/schema/migrations/oss/postgres/35/01_cleanup.up.sql

```diff
@@ -0,0 +1,22 @@
+begin;
+
+-- This adds additional cleanup to what went into 0.8.x. A forgotten case could
+-- mean that sessions can persist in canceled state even if there are no
+-- connections for that session (possibly due to cleanup) and will never be
+-- transitioned because connection state cannot be determined.
+--
+-- There is no reason to transition to terminated state as we remove those
+-- periodically anyways, and there is no useful information to be gleaned from
+-- having it in that state.
```

Comment:
Not transitioning to the terminated state may impact the fact tables in the warehouse. Can we either write sql tests to confirm that deleting will not impact the warehouse? The other option is to not delete them and instead just transition the sessions to the terminated state so the normal workflows continue.